src/control/server/harness.go

```diff
@@ -414,13 +399,12 @@ func (h *IOServerHarness) RestartInstances() error {
 	defer h.RUnlock()
 
 	if !h.IsStarted() {
-		return errors.New("can't start instances: harness not started")
+		return FaultHarnessNotStarted
 	}
-	if !h.IsStartable() {
-		return errors.New("can't start instances: already running")
-	}
-	if len(h.StartedRanks()) > 0 {
-		return errors.New("can't start instances: already started")
+
+	startedRanks := h.StartedRanks()
+	if len(startedRanks) > 0 {
+		return FaultInstancesNotStopped(startedRanks)
```

Comment:
because of the limitation in the current implementation, we have to restart the harness Start() loop which includes restarting dRPC which will break running process. this is on the list of things to be fixed , but not there yet.